### DISCUSSION TITLE
Executions for melee weapons

### DISCUSSION THREAD

**Discussion by Shadowtail117** [HaloSpaceStation/HaloSpaceStation13#2872] [Ideas]
## Summary

Elites can use energy swords/energy daggers to execute downed (read: horizontal) UNSC/URF instead of just bapping them over and over again, leading to a more "cinematic finish". Marines/ODSTs/Spartans could use machetes or a knife to do the same against downed Covenant forces or URF and vice versa.

## Specifics

Right now if someone goes horizontal mid-fight the status quo is to just hit them over and over again with a melee weapon until they ghost or their heart stops or something similar, which seems like a rather anticlimactic end to their life. So, let's add execution-style kills to end someone's life instantly if they're downed and unable to resist. If they're downed, they're probably at the enemy's mercy anyway and could easily be killed using the old method, so it doesn't present a challenge in terms of balancing (outside of armor, more on that later). **Instead, this allows the enemy to know more definitively that someone is dead (and maybe get some satisfaction of the kill), and for the downed player to know that they will not be revived/taken prisoner and are free to ghost**. 

These are NOT the same as assassinations from Halo: Reach and beyond, as the way FOV works in SS13 (and the general fast-paced combat in close quarters) would make such a thing impractical at best and broken at worst. Instead, this would only work on mobs that are fully downed or are aggressive/kill-grabbed by the enemy.

## Implementation

**Everything regarding implementation is moot and subject to change depending on feedback. Items in `code` are especially moot.**

If you are holding an execution-friendly melee weapon (a magnum is not execution-friendly, for example), you can click on a downed player's body on `grab` intent to initiate an execution. `You can target either the chest or head`. Once you do this, your character will get on top of the downed character, along with a visible message in chat and likely some special visual effect, and a do-after will be initiated. `After a couple of seconds,` the execution will "complete" and the downed character will receive fatal damage, `instantly killing them`. If your character is moved, damaged, or interacted with in any way during the execution, it will stop and you will have to try again.

Visible messages for executions vary depending on the weapon, `the targeted zone`, `and the species in question`. `While all of them instantly kill the target,` the exact wound type inflicted would vary; an energy sword is different from a combat knife.

Being horizontal is the only criterion; `the victim can be awake and conscious while being executed,` in which case they'll likely receive a message of their impending doom.

### Allowed species

In the Halo games, only Elites and Spartans are allowed to do executions. Mechanically, because they are the only characters you can play as, but also because they are the strongest, either naturally (in the case of Elites) or through augmentation (in the case of Spartans). However, HSS allows you to play as virtually any race, but it's impractical for an Unggoy to be able to execute a Spartan whose combat knife he picked up from the ground. Thus, only Spartans, Elites, `and ODSTs` are able to perform executions; the rest will have to resort to the old slashing method. As said before, if a player is downed and in the hands of the enemy their life is at their mercy anyway (except in the case of stealth missions, in which case an Unggoy **really** shouldn't be able to execute a Spartan), so this is less-so balancing restrictions with the exception of the parentheses and more-so lore reasons.

### Examples

A marine is downed on the battlefield and dragged away from his comrades by an Elite. The Elite activates his energy sword and clicks on the marine on `grab` intent on his chest. If he weren't already standing over the marine, he will dash to get on top of him and a visible message such as **`Wort 'Wortamee grabs Pvt. Osmo by his neck and prepares to run him through with the energy sword!`** will appear. `After a couple of seconds,` and provided nothing happens, another visible message such as **`Wort 'Wortamee runs Pvt. Osmo through with the energy sword!`** will pop up and the marine will be `instantly killed (code-wise, the brain will instantly die)`. A critical burn wound would appear on the marine's chest to signify his death, while his player is now free to ghost and respawn or do whatever, knowing that his life is definitively over.

A jackal is downed on the battlefield and dragged away by an ODST. He unsheathes his machete and clicks on the jackal on `grab` intent on his head. If he weren't already standing over the jackal, he will dash to get on top of him and a visible message such as **`Lt. Grey steps on Kiq and brandishes the machete!`** will appear. `After a couple of seconds`, and provided nothing happens, another visible message such as **`Lt. Grey slices cleanly through Kiq's neck with the machete!`** will pop up and the jackal will be `instantly killed (code-wise, the brain will instantly die)`. A critical stab wound would appear on the jackal's head (since neck and head are the same) to signify his death, while his player is now free to ghost and respawn or do whatever, knowing his life is definitively over.

**Comment by X0-11** [comment]
isn't this just a fancier throat slit

**Comment by Shadowtail117** [comment]
Related PR is #2873

**Comment by BDpuffy420** [comment]
was added
